docs/gates.md

```diff
@@ -1,58 +1,235 @@
-from cirq.devices import GridQubit
-q0, q1 = (GridQubit(0, 0), GridQubit(0, 1))
-print(CNOT.on(q0, q1))
-print(CNOT(q0, q1))
+import cirq
+
+# This examples uses named qubits to remain abstract.
+# However, we can easily also LineQubits or GridQubits to specify a geometry
+a = cirq.NamedQubit('a')
+b = cirq.NamedQubit('b')
+c = cirq.NamedQubit('c')
+
+# Example Operations, that correspond to the moments above
+print(cirq.H(b))
+print(cirq.CNOT(b, c))
+print(cirq.CNOT(a, b))
+print(cirq.H(a))
+print(cirq.measure(a,b))
 # prints
-# CNOT((0, 0), (0, 1))
-# CNOT((0, 0), (0, 1))
+# H(b)
+# CNOT(b, c)
+# CNOT(a, b)
+# H(a)
+# cirq.MeasurementGate(2, 'a,b', ())(a, b)
 ```
 
-``Gate``s operate on a specific number of qubits and classes that
-implement ``Gate`` must supply either the ``_num_qubits_`` or ``_qid_shape_`` magic method.  For
-convenience one can use the ``SingleQubitGate``, ``TwoQubitGate``,
-and ``ThreeQubitGate`` classes for these common gate sizes.
+This would create the operations needed to comprise the circuit from the above
+diagram.  The next step would be composing these operations into moments and
+circuits.  For more on those types, see the documentation on
+[Circuits](circuits.md).
 
-The most common type of ``Gate`` is one that corresponds to applying
-a unitary evolution on the qubits that the gate acts on.
-``Gate``s can also correspond to noisy evolution on the qubits. This
-version of a gate is not used when sending the circuit to a
-quantum computer for execution, but it can be used with
-various simulators. See [noise documentation](noise.md).
+## Other gate features
 
+Most ``Gate``s operate on a specific number of qubits, which can be accessed
+by the ``num_qubits()`` function.  One notable exception is the
+``MeasurementGate`` which can be applied to a variable number of qubits.
 
-### Magic Methods
+Most gates also have a unitary matrix representation, which can be accessed
+by ``cirq.unitary(gate)``.  
+
+Not all `Gate`s correspond to unitary evolution. They may represent a
+probabilistic mixture of unitaries, or a general quantum channel. The component
+unitaries and associated probabilities of a mixture can be accessed by
+`cirq.mixture(gate)`. The Kraus operator representation of a channel can be
+accessed by `cirq.channel(gate)`. Non-unitary gates are often used in the
+simulation of noise. See [noise documentation](noise.md) for more details.
+
+Many arithmetic operators will work in the expected way when applied to
+gates.  For instance, ``cirq.X**0.5`` represents a square root of X gate.
+These can also be applied to Operators for a more compact representation, such
+as ``cirq.X(q1)**0.5`` will be a square root of X gate applied to the q1
+qubit.  This functionality depends on the "magic methods" of the gate
+being defined (see below for details).
+
+Gates can be converted to a controlled version by using `Gate.controlled()`.
+For instance, `cirq.X.controlled()` returns a `cirq.CNOT` gate.
```

Comment:
This is a special case. I'm concerned that just giving this example might give the false impression. 
In general, `Gate.controlled()` on an arbitrary `Gate` gives a `ControlledGate` except in special circumstances of these well known gates.

Reply:
Done.